### PR TITLE
Move to RedisString (trail 2)

### PR DESCRIPTION
**Pull request by gkorland**
ref https://github.com/RedisJSON/RedisJSON/pull/394

### CODE REVIEW

File: src/commands.rs [hunk 1]
```diff
@@ -486,28 +490,32 @@ pub fn command_json_get<M: Manager>(manager: M, ctx: &Context, args: Vec<String>
...
-            "XX" => set_option = SetOptions::AlreadyExists,
-            "FORMAT" => {
+        match s.try_as_str()? {
+            arg if arg.eq_ignore_ascii_case("NX") => set_option = SetOptions::NotExists,
+            arg if arg.eq_ignore_ascii_case("XX") => set_option = SetOptions::AlreadyExists,
+            arg if arg.eq_ignore_ascii_case("FORMAT") => {
                 format = Format::from_str(args.next_string()?.as_str())?;
             }
-            _ => break,
+            _ => break, // TODO should return an error??
```
Comment (oshadmi) [inline src/commands.rs:514]: Return an error on superfluous arguments - on a different PR?

**Review by oshadmi — APPROVED** [review]
👍🏼 LGTM

File: src/commands.rs [hunk 2]
```diff
@@ -506,12 +506,16 @@ pub fn command_json_set<M: Manager>(
...
     while let Some(s) = args.next() {
         match s.try_as_str()? {
-            arg if arg.eq_ignore_ascii_case("NX") => set_option = SetOptions::NotExists,
-            arg if arg.eq_ignore_ascii_case("XX") => set_option = SetOptions::AlreadyExists,
+            arg if arg.eq_ignore_ascii_case("NX") && set_option == SetOptions::None => {
+                set_option = SetOptions::NotExists
+            }
+            arg if arg.eq_ignore_ascii_case("XX") && set_option == SetOptions::None => {
+                set_option = SetOptions::AlreadyExists
+            }
```
Comment (oshadmi) [inline src/commands.rs:514]: `NX` and `XX` can now occur only once, but `FORMAT` can still occur more times, taking its last value?
(consider adding a `None` value also for FORMAT enum?)

**Review by oshadmi — APPROVED** [review]

File: src/commands.rs (hunk 2 above)
Comment (gkorland) [reply src/commands.rs:514]: @oshadmi I didn't really deal with Format since we don't really support BSON yet, the all FORMAT thing needs a review

**Review by oshadmi — APPROVED** [review]
👍🏼

**Pull request merged**